### PR TITLE
[NIGHTCODING/NOT TESTED] Cobbduceus 5/X: Tox/Oxy TendWound(ies)

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
On draft until tested

### Cleanse Patient

The Tox form of TW, Cleanse patient will slowly remove toxin damage. Upgraded also purges reagents indiscriminately, with Experimental purging JUST harmful chems/ chems with harmful side effects.

Using a beaker instead of your hand will allow you to retrieve the toxin reagent for every bad chem purged (it's a flat 0.1u per iteration).

Tox being the most difficult of the four has the lowest HPS.

### Breathe Life

The Oxy form of TW, Breathe Life will slowly remove oxygen-related damage. Upgrading also removes losebreath.

Ox being the second most has HPS between tox and the B's

### Both

Both have unique branches in the techweb. I've technically made experimental 500 points higher by making it require advanced surgery and both upgrades. Both also have a "femto" or second upgrade post experimental since they're the more difficult damage types.

### Minor

Medbot scaling was adjust appropriately for the 2 new trees (combo is still considered bonus). This means HPS might be lower at some techs than previously but also more levels in between so keep them updated! Also sprinkled some Qdeleted checks where I figured it would be useful since there was runtimes regarding a missing person.

Medbot will now not assess you if you're not next to it AND it's on stationary mode. I'd consider this a fix.

I've typed out the 4 base damages in preparation for a separate organ healing that may or may not be included in this PR.

## Why It's Good For The Game

Ways to allow doctors an active role in healing is good.

## Changelog
:cl: Dr. C.O.BBlestone
add: Cleanse Patient - A Toxin equivalent to Tend Wounds. Includes unique research
add: Breathe Life - An Oxygen equivalent to Tend Wounds. Includes unique research
tweak: Medbot
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Also assumes #46319

**Comment by Shapsy** [comment]
I dunno if there's much point to an oxygen healing surgery since it doesn't really accumulate like the other damage types

**Comment by Swiftkor** [comment]
Seems kind of weird to fix toxin and oxygen damage with surgery, they both seem like damage types that should be kept to chemical solutions. I get that you want to move away from chemistry and make doctors more involved with the healing process but surgery to purge toxins from your system seems especially odd. I feel like you can keep doctors involved with chemical healing by making the application method more complex, which is what you've been doing with your new chemistry system.  In my mind, brute and burn should be solved by surgery primarily with some chemical options while the other two damage types are solved by purely chemical solutions.

**Comment by PKPenguin321** [comment]
2 things
1. Oxyloss healing really doesn't need a full on surgery IMO. It is already arguably the simplest damage type and is quick to heal by just CPRing in a room with breathable atmos. Therefore the cooler option would be something fast and drastic, like incising the neck and inserting a breathing tube. It could be as simple as using a scalpel (or a pen for ghetto) on a downed person that has oxyloss and then jamming in a breathing pipe (we don't really have straws for a ghetto equivalent sadly). Would heal oxyloss very rapidly but would require it to be patched up before it starts causing more oxyloss. Would be neat to use antagonistically too

2. Retrieving chems from a body has historically lead to exploits with unbalanced chems that you aren't supposed to be able to acquire in a beaker being extracted and duplicated, so adding a surgery for it seems iffy to me

**Comment by necromanceranne** [comment]
> Seems kind of weird to fix toxin and oxygen damage with surgery, they both seem like damage types that should be kept to chemical solutions.

I've pushed sorta hard for a toxins surgery specifically just because toxins are a very difficult form of damage to treat at the moment duh to the chemical changes. We do technically have a oxygen loss treatment already. That's CPR. This seems like...just surgerized CPR.

Ideally, like PKP suggested, it'd be neater to have more realistic tools but hell that can come later, this is a good intermediary to fix a ongoing issue.

> 2. Retrieving chems from a body has historically lead to exploits with unbalanced chems that you aren't supposed to be able to acquire in a beaker being extracted and duplicated, so adding a surgery for it seems iffy to me

I think it's not actually directly extracting the chem itself, it's turning it into the toxin reagent, which is hard to acquire but actually is in the recipe for one of Cobby's chems and isn't in of itself powerful at all. It's mostly been used for slimepeople treatment.

**Comment by PKPenguin321** [comment]
Ah, that's cool then

**Comment by Tlaltecuhtli** [comment]
make the toxin tend wounds use leeches

**Comment by dr-bee** [comment]
If there is to be a non-chemical method of healing toxin damge it should be a dialysis machine.

**Comment by shellspeed1** [comment]
> 
> 
> If there is to be a non-chemical method of healing toxin damage it should be a dialysis machine.

Yeah pretty much, treating toxins via surgery is iffy. It might be easier to just have toxins healed by replacing the liver or something if you want a surgical method. idk.

**Comment by Difrit** [comment]
As mentioned in the original post, experimental surgery "JUST harmful chems/ chems with harmful side effects". Does this mean it won't purge OD'd healing chems, which are currently the deadliest thing known to man?

**Comment by Krysonism** [comment]
Toxin surgery is bad game design. 

This is + stasis just makes saving poisoned patients completely trivial and idiot proof.

Cringe

**Comment by CRITAWAKETS** [comment]
The surgery names should be renamed right now because "Cleanse Patient" and "Breathe Life" just imply that you have either opened your 5th eye or you are actually just a diety.

**Comment by Krysonism** [comment]
>The surgery names should be renamed right now because "Cleanse Patient" and "Breathe Life" just imply that you have either opened your 5th eye or you are actually just a diety

I totally agree, Cobby seems to forget that this is a sci-fi game.

**Comment by Mickyan** [comment]
I guess oxy surgery is useful if there's no breathable atmosphere but personally I'd just change CPR to work with internals

**Comment by Shapsy** [comment]
It's science fantasy, see: wizards, lavaland

**Comment by nemvar** [comment]
This doesn't fix https://github.com/tgstation/tgstation/issues/46169 at all. It just silences it. The core of the issue is that you are soft reseting the bot (and nulling the patient in the process) BEFORE you try to remove the trait.

File: code/modules/surgery/healing.dm [hunk 1]
```diff
@@ -1,184 +1,315 @@
...
+	var/antispam = FALSE
+
+
 /datum/surgery_step/heal
 	name = "repair body"
 	implements = list(/obj/item/hemostat = 100, TOOL_SCREWDRIVER = 65, /obj/item/pen = 55)
 	repeatable = TRUE
 	time = 25
+
+/datum/surgery_step/heal/four_damages/
```
Comment (nemvar) [inline code/modules/surgery/healing.dm]: Why exactly a new subtype? Is there another type branching off /datum/surgery_step/heal? Aren't you just making your life harder here?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/healing.dm]: yes/no.

I'm planning on using the heal step for organ damage which will function differently.

If I don't get to it I can easily f/r this back.

**Comment by ExcessiveUseOfCobblestone** [comment]
> The surgery names should be renamed right now because "Cleanse Patient" and "Breathe Life" just imply that you have either opened your 5th eye or you are actually just a diety.
> 
> Also im feeling kinda iffy about toxin damage being healed by surgery. It should be the one thing that is healed by chems mainly. Although having a manual way of removing toxin before reviving someone via surgery could work neatly.

I was picturing the procedure as more of like a spa thing hence why I took out surgical for it.

Breathe Life was going to be something like "open airways", but mobs that intake (not air) would also be candidates for the surgery so I made it sound "cool".

**Comment by ExcessiveUseOfCobblestone** [comment]
> maybe loop like tend wounds, also

they SHOULD since I didn't overwrite the proc.

**Comment by ExcessiveUseOfCobblestone** [comment]
> I dunno if there's much point to an oxygen healing surgery since it doesn't really accumulate like the other damage types

The first one probably isn't, but the second/third drop losebreath which is the accumulation mechanic for oxy.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Seems kind of weird to fix toxin and oxygen damage with surgery, they both seem like damage types that should be kept to chemical solutions. I get that you want to move away from chemistry and make doctors more involved with the healing process but surgery to purge toxins from your system seems especially odd. I feel like you can keep doctors involved with chemical healing by making the application method more complex, which is what you've been doing with your new chemistry system. In my mind, brute and burn should be solved by surgery primarily with some chemical options while the other two damage types are solved by purely chemical solutions.

I like this take

**Comment by Mickyan** [comment]
> they SHOULD since I didn't overwrite the proc.
I meant having CPR loop like tend wounds, it's a bit annoying having to click spam

**Comment by ExcessiveUseOfCobblestone** [comment]
TW abuses the fact the parent proc handles everything but i can check it out

**Pull request closed without merging**